snekbox/nsjail.py

```diff
@@ -227,13 +239,14 @@ def python3(
             # Parse attachments
             try:
                 # Sort attachments by name lexically
-                attachments = sorted(
-                    fs.attachments(self.max_attachments, self.max_attachment_size),
-                    key=lambda a: a.name,
+                attachments = timed(
+                    parse_files,
+                    (fs, self.files_limit, self.files_pattern),
+                    timeout=self.files_timeout,
                 )
                 log.info(f"Found {len(attachments)} attachments.")
-            except AttachmentError as err:
-                log.info(f"Failed to parse attachments: {err}")
+            except TimeoutError as err:
+                log.info(f"Exceeded time limit in parsing attachments: {err}")
                 return EvalResult(args, returncode, f"AttachmentError: {err}")
```

Comment:
The `AttachmentError: ` prefix seems misleading because no such exception type still exists. I don't think it's necessary.

Reply:
Changed in [9400107](https://github.com/python-discord/snekbox/pull/159/commits/9400107e00d37b1809dd1ff8276a6b5bbba9dfdd)